backend/src/ml_space_lambda/utils/iam_manager.py

```diff
@@ -357,6 +357,9 @@ def remove_all_user_roles(self, username: str, projects: List[str]) -> None:
 
     # Creates the IAM role for the MLSpace user/project context
     def _create_iam_role(self, iam_role_name: str, project_name: str, username: str) -> str:
+        tags = generate_tags(username, project_name, self.system_tag)
+        tags.append({"Key": "dynamic-user-role", "Value": "true"})
```

Comment:
Doesn't it make more sense to add this to `generate_tags` as an optional named parameter?

I also feel like having value True is wasting opportunity here. Maybe `{ "role-type": "dynamic-user-role" }`?

I feel like this offers more opportunity for variety in the future.

Reply:
Yeah I didn't love this. I can see accepting a list of additional tags to clean things up a bit.

I'm not as sure about changing the tag though. I get the instinct but since we don't have a use-case for other types of roles it seems potentially limiting (say we wanted roles to be multiple types in the future or something.) I don't think we really lose anything by having separate tags other than having the value be pretty useless.

Reply:
Even if you have hybrid in the future we can just adapt the type naming. Example:
- `dynamic-user-role`
- `dynamic-user-extra-role`

This boils down to preference I think, but I would lean away from creating a pattern that will bloat our tags.

Reply:
I think I'm not going to do this for now (partially out of concern for for the time it would take to change it and the tests) but we can talk about this at another dev sync and always migrate in the same way we're adding these in the first place.